lib/libc/picolibc/Kconfig

```diff
@@ -16,7 +16,7 @@ config PICOLIBC_USE_MODULE
 # force TLS when using toolchain with TLS support
 config PICOLIBC_USE_TOOLCHAIN
 	bool
-	default y if PICOLIBC && !PICOLIBC_USE_MODULE
```

Comment:
I don't think this change is useful, and probably not correct. You only want to use the toolchain picolibc when you're using picolibc.

Reply:
the `PICOLIBC` here should be redundant:

https://github.com/zephyrproject-rtos/zephyr/blob/e912a95355580ed3ae635f02103a6c7a3859611e/lib/libc/picolibc/Kconfig#L4

Reply:
ah, great point. Thanks for providing the wider context. Reviewing patches is always tricky.